### PR TITLE
[DSCP-158] Split lib into core/ witness/ educator/

### PR DESCRIPTION
**Pull request by volhovm**

### CODE REVIEW

File: core/tests/Test/Dscp/Core/Instances.hs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module Test.Dscp.Core.Instances
+    (
+    ) where
```
Comment (Martoon-00) [inline core/tests/Test/Dscp/Core/Instances.hs]: Memno

Comment (volhovm) [reply core/tests/Test/Dscp/Core/Instances.hs]: Fixed.

File: educator/src/Dscp/Educator/Secret/Class.hs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import Dscp.Educator.Secret.Types (EducatorSecret (..))
+import Dscp.Launcher.Rio (RIO)
+
+class MonadEducatorSecret m where
+    -- | Fetch secret key of educator.
+    getEducatorSecret :: m SecretKey
+
+instance HasLens' ctx EducatorSecret =>
+         MonadEducatorSecret (RIO ctx) where
+    getEducatorSecret = views (lensOf @EducatorSecret) esSecretKey
```
Comment (Martoon-00) [inline educator/src/Dscp/Educator/Secret/Class.hs]: Everything under `Dscp.Educator.Secret` is redundant now

Comment (volhovm) [reply educator/src/Dscp/Educator/Secret/Class.hs]: Fixed.

**Pull request merged**